frontend/src/components/auth/Authy.tsx

```diff
@@ -40,40 +50,79 @@ const Authy = ({
     }
   };
 
+  const handlePasscode = (e: { target: { value: unknown } }) => {
+    const newPasscode = e.target.value as string;
+    if (newPasscode !== null && /^[0-9]{0,6}$/.test(newPasscode)) {
+      setPasscode(newPasscode);
+    }
+  };
+
+  const handleFocus = () => {
+    if (inputRef && inputRef.current) {
+      inputRef.current.focus();
+    }
+  };
+
   if (authenticatedUser) {
     return <Redirect to={HOME_PAGE} />;
   }
 
+  const boxIndexes = [0, 1, 2, 3, 4, 5];
+
   if (toggle) {
     return (
-      <div style={{ textAlign: "center" }}>
-        <h1>Authy</h1>
-        <form
-          onSubmit={(event) => {
-            event.preventDefault();
-            onAuthySubmit();
-          }}
-        >
-          <div>
-            <p>Enter token from Authy App</p>
-            <input
-              type="text"
-              value={passcode}
-              onChange={(event) => {
-                setPasscode(event.target.value);
-                setError("");
-              }}
-              placeholder="passcode"
-            />
-          </div>
-          <div>
-            <button className="btn btn-primary" type="submit">
-              Verify
-            </button>
-          </div>
-          {error && <p style={{ color: "red" }}>{error}</p>}
-        </form>
-      </div>
+      <>
+        <Box bg="teal.400" height="100vh" onClick={handleFocus}>
+          <Flex bg="white" height="100vh" width="47%" justifyContent="center" alignItems="center">
+            <VStack width="77%" align="flex-start" gap="3vh">
+              <Text variant="login">One last step!</Text>
+              <Text variant="loginSecondary">In order to protect your account, please enter the authorization code from the Twilio Authy application.</Text>
+              <Flex direction="row" width="100%" justifyContent="space-around">
+                {boxIndexes.map((boxIndex) => {
+                  return (
+                    <Flex
+                      key={boxIndex}
+                      width="56px"
+                      height="56px"
+                      border="1px solid #A7A7A7"
+                      borderRadius="4px"
+                      justify="center"
+                      align="center"
+                    >
+                      <Text variant="authyDigit" textAlign="center">
+                        {passcode.length > boxIndex ? passcode[boxIndex] : " "}
+                      </Text>
+                    </Flex>
+                  )
+                })}
+              </Flex>
+              <Button
+                variant="login"
+                disabled={passcode.length < 6}
+                _hover={
+                  email && password
```

Comment:
just curious since i don't know the inner workings of the signup stuff, but doesn't being on this page imply that the user entered an email and password? so why is it being checked here?

Reply:
It is just cause I copied the button from login, it does not need to check email and password. I'll change it.